typescript/src/deposit.ts

```diff
@@ -67,15 +67,51 @@ export interface Deposit {
 /**
```

Comment:
The presented solution should work but has severe flaws regarding code cleanliness:
- It complicates the domain types declared in `deposit.ts` by adding new ones that are here just to capture the caveats of a single field (`refundPublicKey`). That complicates the entire library and makes it less generic just to conform to a single use case, i.e. the ability to pass the refund public key as a hash from the dApp.
- Introduces ambiguous functions, i.e. `assembleDepositScript` and `buildDepositScript`. Their names say nothing about the differences between them and which one should be used in which context.

I think we should change the approach. In the beginning, we should note that the `DepositScriptParameters` is a specialized type that is used only during deposit script construction, and `walletPublicKey` and `refundPublicKey` are not used as plain-text public keys in this context but always as public key hashes. That means we can change `DepositScriptParameters` to be:
```
export type DepositScriptParameters = Pick<
  Deposit,
  | "depositor"
  | "blindingFactor"
  | "refundLocktime"
> & {
    /** Documentation */
    walletPublicKeyHash: string

    /** Documentation */
    refundPublicKeyHash: string 
}
```

Once we do that, we gain the possibility to pass PKH's directly to the `calculateDepositAddress` function. On the other hand,  we can no longer pass `Deposit` as `DepositScriptParameters` in the places we did it so far. However, this can be easily fixed using `computeHash160` to convert the plain-text public key to a public key hash where applicable. The presented solution can involve more changes in the codebase but is much cleaner and should keep the library API consistent. What do you think @michalsmiarowski ?

Reply:
I agree! Now I also think that my solution introduces unnecessary complications just for a single, pretty simple, use and even good comments won't compensate that as it still might be initially confusing.

Fixing that right now!